### DISCUSSION TITLE
import only the named tiddlers or the tiddlers matching a filter from an external wiki - single file or node.

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#5908] [Ideas]
Perhaps the new JSON store area may help, but would it be possible to have actions that once triggered would import only the named tiddlers or the tiddlers matching a filter from an external wiki - single file or node. 

This would be a pull method to get content from wikis to which you have access. I understand that with single file wikis the whole wiki needs to be read, but the issue here is about how it looks to the user of the wiki doing the import. A one way solution is already a great tool. Remember I can do this manually export/import, drag and drop I just want to bundle it in a way a user need not be bothered with the mechanics below.

If two wikis can pull content from each other a two way communication can occur, but I am not asking for a universal solution, but one a tiddlywiki designer can make use of through choosing and managing the tiddler namespaces and filters, currency and contention themselves. 

To me this would be a simpler version of federation

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio under Node.js you can use the "fetch" command to retrieve another single file TiddlyWiki, and import the tiddlers selected by a filter. It also lets you specify a filter that transforms the titles (eg to add a prefix).

Sadly, we cannot offer the same thing in the single file edition because of our old nemesis CORS, which prevents JavaScript from retrieving the HTML content of other websites (except in special circumstances specified by CORS).

> To me this would be a simpler version of federation

The `--fetch` command is actually the primitive that enables federation under Node.js. The problem has always been that we can't offer the same functionality in the browser.
